website/src/pages/tech-specs.md

```diff
@@ -0,0 +1,371 @@
+
+- **files** - Partition path to file name index. Key for the Hudi record is the partition path and the actual record is a map of file name to an instance of [HoodieMetadataFileInfo](https://github.com/apache/hudi/blob/master/hudi-common/src/main/avro/HoodieMetadata.avsc#L34) . The files index can be used to do file listing and do filter based pruning of the scanset during query
+
+- **bloom_filters** - Bloom filter index to help map a record key to the actual file. The Hudi key is `hash(partition name) + hash(file name)` and the actual payload is an instance of [HudiMetadataBloomFilter](https://github.com/apache/hudi/blob/master/hudi-common/src/main/avro/HoodieMetadata.avsc#L66) . Bloom filter helps with identifying the files to update and optimize joins with row key and point queries with row key filter.
+- **column_stats** - contains statistics of all the columns for all the rows in the table. This enables file grained file pruning for filters and join conditions in the query. The actual payload is an instance of [HoodieMetadataColumnStats](https://github.com/apache/hudi/blob/master/hudi-common/src/main/avro/HoodieMetadata.avsc#L101). 
+
+
+
+## Data Organization
+
+As mentioned in the data model, data is partitioned coarsely through a directory hierarchy based on the partition path configured. Within each partition the data is physically stored as **base and log files** and organized into logical concepts as **File groups and File slices**. The logical concepts will be referred by the writer / reader requirements. 
+
+**File group** - Groups multiple versions of a base file. File group is uniquely identified by a File id. Each version corresponds to the commit instants timestamp recording updates to rows in the file. The base file are stored in open source data formats like  Apache Parquet, Apache ORC, Apache HBase HFile etc.
+
+**File slice** - A File group can further be split into multiple files. A base file corresponding to commit timestamp and a set of log files that batches the deletes/updates to the base file and each log file corresponds to a delta commit timestamp and the delta commits happen after the base file commit in the Hudi timeline. 
+
+
+
+### **Base file**
+
+The base file name format is:
+
+```
+[File Id]_[File Write Token]_[Transaction timestamp].[File Extension]
+```
+
+- **File Id** - Uniquely identify a base file within a partition. Multiple versions of the base file share the same file id.
+- **File Write Token** - Monotonically increasing token for every attempt to write the base file. This should help uniquely identifying the base file when there are failures and retries. Cleaner can clean-up partial base files if the write token is not the latest in the file group 
```

Comment:
Worth calling out that monotonicity is only provided w/in the same Query Engine (for ex, Spark) session

Reply:
We can draw a picture like below:

![image](https://user-images.githubusercontent.com/7644508/183544138-641a6e1a-c5f1-48a8-ac18-6733af51e75a.png)

copied from one of my talk

Reply:
We just describe what is the format of the base file naming here. I think since we specify that this field is used for retries - it is clear that this is useful within the same engine context?

Reply:
Makes sense, but the framing of the first sentence describes it very generically, while you're referring to only Spark Task/Stage retries:

> Monotonically increasing token for every attempt to write the base file.